### PR TITLE
Reuse NATS connection to enhance NATS publisher performance

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Reuse NATS connection to enhance NATS publisher performance.

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/10722

### CODE REVIEW

**Review by montaro — DISMISSED** [review]

File: components/event-publisher-proxy/pkg/sender/nats.go [hunk 1]
```diff
@@ -19,14 +20,14 @@ type GenericSender interface {
 
 // HttpMessageSender is responsible for sending messages over HTTP.
 type NatsMessageSender struct {
-	Ctx    context.Context
-	Target string
-	Logger *logrus.Logger
+	Ctx        context.Context
```
Comment (sayanh) [inline components/event-publisher-proxy/pkg/sender/nats.go:23]: Do `Ctx` and `Logger` need to be public?

File: components/event-publisher-proxy/pkg/sender/nats_test.go [hunk 2]
```diff
@@ -9,89 +9,50 @@ import (
...
 
+	pkgnats "github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/nats"
 	testingutils "github.com/kyma-project/kyma/components/event-publisher-proxy/testing"
-	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
-	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers"
-	eventingtesting "github.com/kyma-project/kyma/components/eventing-controller/testing"
 )
 
-func TestNatsSender(t *testing.T) {
+func TestNatsSender2(t *testing.T) {
```
Comment (sayanh) [inline components/event-publisher-proxy/pkg/sender/nats_test.go:19]: `TestNatsSender2` doesn't imply anything. Can we add some context in the function name or use something like `t.Run("...some context...")`

File: components/event-publisher-proxy/pkg/sender/nats.go (hunk 1 above)
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/sender/nats.go]: Done.

File: components/event-publisher-proxy/pkg/sender/nats_test.go (hunk 2 above)
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/sender/nats_test.go]: Done.

**Review by sayanh — APPROVED** [review]

**Review by k15r — APPROVED** [review]
lgtm

**Review by sayanh — APPROVED** [review]

**Pull request merged**